mne/preprocessing/maxwell.py

```diff
@@ -108,24 +110,29 @@ def maxwell_filter(raw, origin=(0., 0., 40.), int_order=8, ext_order=3,
     st_corr = float(st_corr)
     if st_corr <= 0. or st_corr > 1.:
         raise ValueError('Need 0 < st_corr <= 1., got %s' % st_corr)
-    logger.info('Bad channels being reconstructed: ' + str(raw.info['bads']))
+    logger.info('Maxwell filtering raw data')
+    if len(raw.info['bads']) > 0:
+        logger.info('    Bad channels being reconstructed: %s'
+                    % raw.info['bads'])
+    else:
+        logger.info('    No bad channels')
+    raw_sss = raw.copy().load_data(verbose=False)
+    del raw
 
     # If necessary, load fine calibration and overwrite sensor geometry
     if fine_cal is not None:
-        cal_chans = _read_fine_cal(raw.info, fine_cal)
-        raw.info = _update_sensor_geometry(raw.info, cal_chans)
+        logger.info('    Using fine calibration %s' % op.basename(fine_cal))
+        cal_chans = _read_fine_cal(raw_sss.info, fine_cal)
```

Comment:
this needed to be done on `raw_sss.info` not `raw.info`, since it operates in-place